rollup-interface/src/state_machine/da.rs

```diff
@@ -56,14 +54,54 @@ pub trait DaVerifier {
     ) -> Result<(), Self::Error>;
 }
 
+// Simple structure that implements a buffer that counts the number of bytes read from the beginning
+// Useful for the partial blob reading optimization: we know for each blob how many bytes have been read from the beginning
+pub struct BufWithCounter<B: Buf> {
+    inner: B,
+    counter: usize,
+}
+
+impl<B: Buf> BufWithCounter<B> {
+    pub fn new(inner: B) -> Self {
+        BufWithCounter { inner, counter: 0 }
+    }
+
+    // To use with care: if any Buf method is applied to inner directly the counter won't be updated
```

Comment:
This method consumes `BufWithCounter ` so there is no more counter after the method is called.

Reply:
Solved with the last commit